include/unifex/detail/concept_macros.hpp

```diff
@@ -0,0 +1,347 @@
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_requires(...) \
+    requires(__VA_ARGS__) UNIFEX_CONCEPT_FRAGMENT_REQS_
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_M(REQ) \
+    UNIFEX_PP_CAT2(UNIFEX_CONCEPT_FRAGMENT_REQS_M, UNIFEX_PP_IS_PAREN(REQ))(REQ);
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_requires(...) \
+    requires __VA_ARGS__
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_typename(...) \
+    typename UNIFEX_PP_EAT_TYPENAME_(__VA_ARGS__)
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_noexcept(...) \
+    { __VA_ARGS__ } noexcept
+
+  #define UNIFEX_FRAGMENT(NAME, ...) \
+    NAME<__VA_ARGS__>
+
+#else
+
+  #define UNIFEX_CONCEPT UNIFEX_INLINE_VAR constexpr bool
+
+  #define UNIFEX_CONCEPT_FRAGMENT(NAME, ...) \
+    auto NAME ## UNIFEX_CONCEPT_FRAGMENT_impl_ \
+      UNIFEX_CONCEPT_FRAGMENT_REQS_ ## __VA_ARGS__> {} \
+    template <typename... As> \
+    char NAME ## UNIFEX_CONCEPT_FRAGMENT_( \
+      ::unifex::_concept::tag<As...> *, \
+      decltype(&NAME ## UNIFEX_CONCEPT_FRAGMENT_impl_<As...>)); \
+    char (&NAME ## UNIFEX_CONCEPT_FRAGMENT_(...))[2] \
+    /**/
+  #define M(ARG) ARG,
+  #if defined(_MSC_VER) && !defined(__clang__)
+    #define UNIFEX_CONCEPT_FRAGMENT_TRUE(...) \
+      ::unifex::_concept::true_<decltype( \
+        UNIFEX_PP_FOR_EACH(UNIFEX_CONCEPT_FRAGMENT_REQS_M, __VA_ARGS__) \
+        void())>()
+  #else
+    #define UNIFEX_CONCEPT_FRAGMENT_TRUE(...) \
+      !(decltype(UNIFEX_PP_FOR_EACH(UNIFEX_CONCEPT_FRAGMENT_REQS_M, __VA_ARGS__) \
+      void(), \
+      false){})
+  #endif
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_requires(...) \
+    (__VA_ARGS__) -> std::enable_if_t<UNIFEX_CONCEPT_FRAGMENT_REQS_2_
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_2_(...) \
+    UNIFEX_CONCEPT_FRAGMENT_TRUE(__VA_ARGS__)
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_M(REQ) \
+    UNIFEX_PP_CAT2(UNIFEX_CONCEPT_FRAGMENT_REQS_M, UNIFEX_PP_IS_PAREN(REQ))(REQ),
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_requires(...) \
+    ::unifex::requires_<__VA_ARGS__>
+  #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_typename(...) \
+    static_cast<::unifex::_concept::tag<__VA_ARGS__> *>(nullptr)
+  #if defined(__GNUC__) && !defined(__clang__)
+    // GCC can't mangle noexcept expressions, so just check that the
+    // expression is well-formed.
+    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70790
+    #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_noexcept(...) \
+      __VA_ARGS__
+  #else
+    #define UNIFEX_CONCEPT_FRAGMENT_REQS_REQUIRES_noexcept(...) \
+      ::unifex::requires_<noexcept(__VA_ARGS__)>
+  #endif
+
+  #define UNIFEX_FRAGMENT(NAME, ...) \
+    (1u==sizeof(NAME ## UNIFEX_CONCEPT_FRAGMENT_( \
+      static_cast<::unifex::_concept::tag<__VA_ARGS__> *>(nullptr), nullptr)))
+
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+// UNIFEX_TEMPLATE
+// Usage:
+//   UNIFEX_TEMPLATE(typename A, typename B)
+//     (requires Concept1<A> && Concept2<B>)
```

Comment:
Is this supposed to be `(requires Concept1<A> UNIFEX_AND Concept2<B>)`?

Reply:
Yes.